### PR TITLE
configurable date format and handle decimal InvalidOperation

### PR DESCRIPTION
**Pull request by aroder**
- Adds a property `date_format` that makes the target key's date format configurable. It defaults to the original hard-coded date format, making this change backwards compatible.
- Adds a line to adjust the decimal precision if it is too small compared to the tap's schema

@ome9ax are you open to a PR for this, or open to discuss?

### CODE REVIEW

**Comment by aroder** [comment]
Hi @ome9ax bumping this. Do you have time to review?

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
 jsonschema==3.2.0
 boto3==1.18.22
 backoff==1.11.1
+pytest-cov
+moto[s3]
```
Comment (ome9ax) [inline requirements.txt:5]: Those are already in the `[options.extras_require]` section L34-35. I keep them apart as they are not needed to run the package, only to run the tests.
My rational is to keep the target lean if possible.

So I don't know how much of a best practice it is but I used
```python
pip install --upgrade .[test,lint,dist]
```
To install what's needed on `local`. It was my way of following the KISS principle here.

File: target_s3_jsonl/__init__.py [hunk 2]
```diff
@@ -13,12 +13,13 @@
 
 from jsonschema import Draft4Validator, FormatChecker
 from decimal import Decimal
+from adjust_precision_for_schema import adjust_decimal_precision_for_schema
```
Comment (ome9ax) [inline target_s3_jsonl/__init__.py:16]: That a great one. Thanks for adding it.

**Review by ome9ax — COMMENTED** [review]
Thanks for your contribution @aroder
Left a note, Let me know your thoughts.

File: target_s3_jsonl/__init__.py (hunk 2 above)
Comment (ome9ax) [reply target_s3_jsonl/__init__.py:16]: Reading the [Handle multipleOf overflow](https://github.com/python-jsonschema/jsonschema/pull/746) fix introduced in [jsonschema v4.0.0](https://github.com/python-jsonschema/jsonschema/releases/tag/v4.0.0), I wonder if that update doesn't make `adjust_precision_for_schema` reduntant?

**Comment by haleemur** [comment]
I would like to note that the current implementation allows for specifying the date format already. the additional parameter might be redundant.

e.g.

```
naming_convention: "{stream}/export_date={date:%Y-%m-%d}/{timestamp}-{stream}.json"
```

however, we should add some parsing logic to ensure that the naming convention provided is valid and update the documentation with some examples.

**Comment by ome9ax** [comment]
`adjust-precision-for-schema` merged in #75 and available in the `v1.2.2` release @aroder .
Also closes #69 .
I believe this PR can now be closed unless reasons not to.

**Comment by ome9ax** [comment]
All the functionalities are now implemented.

**Pull request closed without merging**